### ISSUE TITLE
./build_and_test.sh fails (Ubuntu Xenial)

### ISSUE THREAD

**Issue by tcf909** [inotify-tools/inotify-tools#114]
Hello,

I can't seem to get the latest master to to build and test on a fresh Ubuntu Xenial docker image.

The result of:

```
./build_and_test.sh
```
is:

```
....
ok 1 - Exit code 0 is returned
ok 1 - Exit code 2 is returned
# passed all 1 test(s)
1..1
# passed all 1 test(s)
1..1
not ok 1 - event logged
ok 1 - Exit code 2 is returned
ok 1 - event logged
#
#	    run_ &&
#	    grep ATTRIB $logfile
#
# passed all 1 test(s)
1..1
# passed all 1 test(s)
1..1
# failed 1 among 1 test(s)
1..1
Makefile:38: recipe for target 'inotifywait-daemon-logs-chown.t' failed
make[1]: *** [inotifywait-daemon-logs-chown.t] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: Leaving directory '/inotify-tools-orig/t'
Makefile:31: recipe for target 'test' failed
make: *** [test] Error 2
```

Hoping to get to the bottom of this so I can write some other tests.

Thanks

**Comment by tcf909** [comment]
And if I try ./build_and_test.sh against the latest 3.20.2.2 tag I get:

```
libinotifytools/src/Makefile.am: installing 'config/depcomp'
parallel-tests: installing 'config/test-driver'
checking whether make sets $(MAKE)... yes
checking for gcc... clang
checking whether the C compiler works... no
configure: error: in `/inotify-tools-orig':
configure: error: C compiler cannot create executables
See `config.log' for more details
```

Any guidance is appreciated.

**Comment by tcf909** [comment]
```
root@c70a0d54c781:/inotify-tools-orig# gcc -v
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-linux-gnu/5/lto-wrapper
Target: x86_64-linux-gnu
Configured with: ../src/configure -v --with-pkgversion='Ubuntu 5.4.0-6ubuntu1~16.04.12' --with-bugurl=file:///usr/share/doc/gcc-5/README.Bugs --enable-languages=c,ada,c++,java,go,d,fortran,objc,obj-c++ --prefix=/usr --program-suffix=-5 --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --libdir=/usr/lib --enable-nls --with-sysroot=/ --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --with-default-libstdcxx-abi=new --enable-gnu-unique-object --disable-vtable-verify --enable-libmpx --enable-plugin --with-system-zlib --disable-browser-plugin --enable-java-awt=gtk --enable-gtk-cairo --with-java-home=/usr/lib/jvm/java-1.5.0-gcj-5-amd64/jre --enable-java-home --with-jvm-root-dir=/usr/lib/jvm/java-1.5.0-gcj-5-amd64 --with-jvm-jar-dir=/usr/lib/jvm-exports/java-1.5.0-gcj-5-amd64 --with-arch-directory=amd64 --with-ecj-jar=/usr/share/java/eclipse-ecj.jar --enable-objc-gc --enable-multiarch --disable-werror --with-arch-32=i686 --with-abi=m64 --with-multilib-list=m32,m64,mx32 --enable-multilib --with-tune=generic --enable-checking=release --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=x86_64-linux-gnu
Thread model: posix
gcc version 5.4.0 20160609 (Ubuntu 5.4.0-6ubuntu1~16.04.12)
```

**Comment by ericcurtin** [comment]
`sudo apt install clang` is what you want :)

We don't use gobjc compiler

**Comment by tcf909** [comment]
```
apt install clang
```
Seems to fix this issue

**Comment by tcf909** [comment]
Lol. Jinx.

**Comment by tcf909** [comment]
Ok, so actually, now back to branch of master:

```
ok 1 - Exit code 2 is returned
# passed all 1 test(s)
1..1
ok 1 - Exit code 0 is returned
# passed all 1 test(s)
1..1
ok 1 - event logged
not ok 1 - event logged
ok 1 - Exit code 2 is returned
#
#	    run_ &&
#	    grep ATTRIB $logfile
#
# passed all 1 test(s)
1..1
# passed all 1 test(s)
1..1
# failed 1 among 1 test(s)
1..1
Makefile:38: recipe for target 'inotifywait-daemon-logs-chown.t' failed
make[1]: *** [inotifywait-daemon-logs-chown.t] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: Leaving directory '/inotify-tools/t'
Makefile:31: recipe for target 'test' failed
make: *** [test] Error 2
```

**Comment by ericcurtin** [comment]
But does the `inotifywait-daemon-logs-chown.t` test pass for you now? We test on Xenial on our build server and it passes. I wrote that test to close this issue #112

Just make sure you're running the master version of the code.

What filesystem are you using?

Is there an older version of inotify-tools or it's library installed that this test could be using by mistake?

I'd also try changing the sleep to 2 in that test and the timeout to 4, to see if that fixes the issue. If that works feel free to push that to this repo as a fix :)

Passes on all the build servers and my personal Xenial system, but all systems are different

**Comment by ericcurtin** [comment]
What's more important is to get a passing test for your new cool feature, you can just run a single test if you want (you're new one for example).

**Comment by tcf909** [comment]
I'm using the latest docker ubuntu:xenial tag.

Definitely want to get my test passing, but want to make sure I can have a "normal" build or I'll be running around in circles debugging.

**Comment by tcf909** [comment]
Also the "trash" directory has two empty files "log" and "test-file". Surprised to find "log" empty.

**Comment by ericcurtin** [comment]
Hmmm I'll try a docker version on my system I'll let you know. It might not be a bad idea to commit a dockerfile to this repo for new users.

That file is intentionally empty because the test only chowns the file it to see if inotify can pick up the ownership change.

**Comment by tcf909** [comment]
Manually watching the file using the tests libraries, binary, etc seems to work ok. Will continue to troubleshoot.

Thanks for the involvement.

**Comment by ericcurtin** [comment]
Yeah I can confirm it fails on my container on my machine too, don't worry about that test, skip it. I'll fix it myself in a day or two.

**Comment by tcf909** [comment]
ok

**Comment by ericcurtin** [comment]
You're right though the file log should have something contained in it, the ATTRIB thing. Anyway I fixed it I believe, USER environment variable seems to be blank in docker container so we now use the user id from $(id -u) instead latest commit on master should fix it 8672a5e26237555dcf6c3f009e63066b04b680b5

You can test also if you'd like and I'll close this then :)
